### PR TITLE
Provide How-to guide for installing a Debian package via Software Update

### PR DESCRIPTION
**Pull request by gabriela-yoncheva**
[#12] Provide How-to guide for Software Update

- created an application that spins up a hello Debian package at the edge
- added a guide to the Eclipse Kanto Documentation
- updated Software Update concept page with link to the how to guide
- created bash script for installing hello Debian package

Signed-off-by: Gabriela Yoncheva <Gabriela.Yoncheva@bosch.io>

### CODE REVIEW

File: quickstart/hono_commands_su.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+software_update_action_template = Template("""
+{
+    "softwareModules": [{
+          "softwareModule": {
+            "name": "$module_name",
+            "version": "$version"
+          },
+          "artifacts": [{
+            "checksums": {
+              "SHA1": "$sha1"
```
Comment (e-grigorov) [inline quickstart/hono_commands_su.py:48]: SHA256 should be a better option.

File: quickstart/hono_commands_su.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            "download": {
+              "HTTPS": {
+                "url": "$url"
+              }
+            },
+            "filename": "install.sh",
+            "size": 	$size
+          }]
+        }],
+        "correlationId": "$correlation_id"
```
Comment (e-grigorov) [inline quickstart/hono_commands_su.py:59]: I would suggest to format the template as typical JSON. correlationId is on the same level as softwareModules.

File: web/site/content/docs/getting-started/hono.md [hunk 3]
```diff
@@ -27,6 +27,7 @@ they communicate remotely with Eclipse Hono only. To run them, you need:
   wget https://github.com/eclipse-kanto/kanto/raw/main/quickstart/hono_commands.py && \
   wget https://github.com/eclipse-kanto/kanto/raw/main/quickstart/hono_events.py && \
   wget https://github.com/eclipse-kanto/kanto/raw/main/quickstart/requirements.txt && \
+  wget https://github.com/eclipse-kanto/kanto/raw/main/quickstart/hono_update_software.py && \
```
Comment (e-grigorov) [inline web/site/content/docs/getting-started/hono.md:30]: The script is called: hono_comments_su.py and should be downloaded in the related how-to guide document.

File: web/site/content/docs/how-to-guides/update-software.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+```shell
+python3 hono_events.py -t demo
+```
+
+In another terminal, we are ready to spin up a `hello` Debian package at the edge via executing the second application
+that requires the Eclipse Hono tenant (`-t`) and the device identifier (`-d`):
+
+```shell
+python3 hono_update_software.py -t demo -d demo:device
```
Comment (e-grigorov) [inline web/site/content/docs/how-to-guides/update-software.md:54]: The file name is not up to date.

**Review by e-grigorov — CHANGES_REQUESTED** [review]

File: quickstart/hono_commands_su.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+import threading
+import time
+import uuid
+from string import Template
+
+from proton import Message
+from proton.handlers import MessagingHandler
+from proton.reactor import Container
+
+ditto_live_inbox_msg_template = Template("""
```
Comment (e-grigorov) [inline quickstart/hono_commands_su.py:26]: As we have a dedicated software update script, we don't need this generic template. Both templates can be merged and it'll remove a lot of inlined strings.

File: quickstart/hono_commands_su.py [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+    def on_start(self, event):
+        conn = event.container.connect(self.server, sasl_enabled=True, allowed_mechs="PLAIN", allow_insecure_mechs=True,
+                                       user="consumer@HONO", password="verysecret")
+        event.container.create_sender(conn, self.address)
+
+    def on_sendable(self, event):
+        print('[sending command]')
+        correlation_id = str(uuid.uuid4())
+        namespaced_id = device_id.split(':', 1)
+        action = "install"
```
Comment (e-grigorov) [inline quickstart/hono_commands_su.py:101]: As it's a fixed value, it can be inlined in the template.

File: quickstart/hono_commands_su.py [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+        feature = "SoftwareUpdatable"
```
Comment (e-grigorov) [inline quickstart/hono_commands_su.py:102]: As it's a fixed value, it can be inlined in the template.

File: quickstart/hono_commands_su.py [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+
+        value = json.dumps(json.loads(software_update_action_template.substitute(
+            module_name="hello", version="2.10", size=30,
```
Comment (e-grigorov) [inline quickstart/hono_commands_su.py:105]: All fixed values can be inlined in the template.

File: quickstart/hono_commands_su.py [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+            url="https://github.com/eclipse-kanto/kanto/raw/main/quickstart/install_hello.sh",
```
Comment (e-grigorov) [inline quickstart/hono_commands_su.py:106]: * I would suggest to use "install-hello" for the module name as this module is about the install script.
* If the module is about the install script, the version can be: 1.0.0

File: quickstart/hono_commands_su.py [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+    response_handler.stop()
+    thread.join(timeout=5)
+    print('[stopped]')
+    exit(0)
+
+
+signal.signal(signal.SIGINT, handler)
+while True:
+    pass
+
```
Comment (e-grigorov) [inline quickstart/hono_commands_su.py:159]: Redundant new line.

**Review by e-grigorov — CHANGES_REQUESTED** [review]

File: web/site/content/docs/how-to-guides/update-software.md [hunk 11]
```diff
@@ -19,14 +19,14 @@ script will be downloaded and executed.
 To ensure that your edge device is capable to execute the steps in this guide, you need:
 
 * Debian-based linux distribution and the `apt` command line tool
-* If you don't have an installed and running Eclipse Kanto, follow [Install Eclipse Kanto]({{< relrefn "install" >}})
+* If you don't have an installed and running Eclipse Kanto, follow {{% relrefn "hono" %}} Install Eclipse Kanto {{% /relrefn %}}
```
Comment (e-grigorov) [inline web/site/content/docs/how-to-guides/update-software.md:22]: The link should point to the Kanto installation, not explore via Hono.

**Review by e-grigorov — CHANGES_REQUESTED** [review]

**Review by konstantina-gramatova — APPROVED** [review]

**Pull request merged**